pkg/terraformer/terraformer.go

```diff
@@ -111,6 +111,16 @@ func (t *Terraformer) execute(command Command) (rErr error) {
 		}
 	}()
 
+	// Sometimes a state is empty because the Terraformer could never run successfully.
+	// Hence, we take a shortcut here and just remove the finalizer.
+	emptyState, err := t.isStateEmpty(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to check if state is empty: %w", err)
+	}
```

Comment:
You can do this check only if the command is destroy, this way one read request will be saved.